website/content/docs/concepts/topologies.md

```diff
@@ -42,6 +51,156 @@ lifecycle of a topology, which typically goes through the following stages:
+
+Here's the corresponding Java code for the processing logic shown in the diagram:
+
+```java
+package heron.dsl.example;
+
+import com.twitter.heron.dsl.*;
+import com.twitter.heron.dsl.impl.BaseStreamlet;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+public final class ExampleDSLTopology {
+    public ExampleDSLTopology() {}
+
+    public static void main(String[] args) {
+        Streamlet<Integer> randoms = BaseStreamlet.createSupplierStreamlet(() -> ThreadLocalRandom.current().nextInt(1, 11));
+        Streamlet<Integer> zeroes = BaseStreamlet.createSupplierStreamlet(() -> 0);
+
+        Builder builder = Builder.CreateBuilder();
+
+        builder.newSource(zeroes)
+                .setName("zeroes");
+
+        builder.newSource(randoms)
+                .setName("random-ints")
+                .map(i -> i + 1)
+                .setName("add-one")
+                .union(zeroes)
+                .setName("unify-streams")
+                .filter(i -> i != 2)
+                .setName("remove-all-twos")
+                .log();
+
+        Config conf = new Config();
+        conf.setNumContainers(2);
+
+        new Runner().run("ExampleDSLTopology", conf, builder);
+    }
+}
+```
+
+This Java code will produce this [physical plan]():
+
+![Heron DSL physical plan](/img/dsl-physical-plan.png)
+
+### Key-value streamlets
+
+In order to perform some operations, such as streamlet joins and streamlet reduce operations,
+
+### Heron DSL topologies
+
+With the Heron DSL *you still create topologies*, but only implicitly. Heron automatically
+performs the heavy lifting of converting the streamlet-based processing logic that you
+create into spouts and bolts and, from there, into containers that are then deployed using
+whichever scheduler your Heron cluster is using.
+
+### Why a DSL?
+
+The older, spouts-and-bolts based model of creating topologies provides 
+
+### Available DSLs
+
+The Heron DSL is currently available in the following languages:
+
+* [Python](../../developers/python/python-dsl)
+
+## Partitioning
+
+## Windowing
+
+Window configurations have two core components:
+
+1. The length of the window
+
+> In the Heron DSL, all window lengths and sliding intervals are specified **in seconds**.
+
+### Sliding windows
+
+In a sliding time window, tuples
+
+![Sliding time window](https://www.lucidchart.com/publicSegments/view/57d2fcbb-591b-4403-9258-e5b8e1e25de2/image.png)
+
+### Tumbling windows
+
+> One of the crucial differences between sliding and tumbling windows is that sliding
+> windows can overlap.
+
+![Tumbling time window](https://www.lucidchart.com/publicSegments/view/881f99ee-8f93-448f-a178-b9f72dce6491/image.png)
+
+### Count windows
+
+There are thus four total window types:
+
+## Resource allocation with the Heron DSL
+
+Three configs:
+
+1. Number of containers
+1. CPU
+1. RAM
+
+The defaults:
+
+Resource | Default
+:--------|:-------
+Number of containers | 1
+CPU | 1.0
+RAM | 512 MB
+
```

Comment:
Can we talk about how Resources can be specified?

Reply:
Again, this is a work in progress. I'll ping you as soon as it's in reviewable shape.